bluefog/torch/optimizers.py

```diff
@@ -305,23 +314,16 @@ def __init__(self, params, model, reduce_type, num_steps_per_communication=1):
 
         self._models = models
         self._parameter_names = {v: k for k, v in sorted(named_parameters)}
+        self._name_parameters = {k: v for k, v in sorted(named_parameters)}
```

Comment:
Seems that only k is used. Can we use a set instead to save some memory?

Reply:
It should not be a big deal because the parameter in python is just a reference key to the tensor.